extensions/icd/actions/tfu/icd.ts

```diff
@@ -0,0 +1,176 @@
+import { type Setting, type Action } from '@awell-health/extensions-core'
+import { Category } from '@awell-health/extensions-core'
+import axios from 'axios'
+import { isEmpty, isNil } from 'lodash'
+import z from 'zod'
+import { FieldSchema, fields } from './config/fields'
+import { dataPoints } from './config/dataPoints'
+
+const settings = {} satisfies Record<string, Setting>
+const SettingSchema = z.unknown()
+
+const PayloadSchema = z.object({
+  fields: FieldSchema,
+  settings: SettingSchema,
+})
+
+export const icd: Action<
+  typeof fields,
+  typeof settings,
+  keyof typeof dataPoints
+> = {
+  key: 'icd',
+  category: Category.DEMO,
+  title: 'ICD-10 Code Lookup',
+  description: 'Lookup ICD-10 codes and return the parent disease',
+  fields,
+  previewable: true,
+  dataPoints,
+  onActivityCreated: async (payload, onComplete): Promise<void> => {
+    const {
+      fields: { icd_codes },
+    } = PayloadSchema.parse(payload)
+    // if no codes, just return blank
+    if (icd_codes.length === 0) {
+      await onComplete({
+        data_points: {
+          codes: JSON.stringify([]),
+          stringResponse: 'Blank',
+        },
+      })
+      return
+    }
+
+    const accessToken = await grabAccessToken()
+
+    const icdCodes = await grabICDCodes(accessToken, icd_codes)
+
+    const resp = icdCodes.some(checkICDCode)
+
+    await onComplete({
+      data_points: {
+        codes: JSON.stringify(icdCodes),
+        stringResponse: resp ? 'TFU' : 'TCM',
```

Comment:
I would rather prefer to retrieve the codes and name of the disease and not if it is TFU or TCM. That is a lot less flexible. Why did we decide to go this approach?